horace_core/sqw/@DnDBase/remove_legacy_alignment.m

```diff
@@ -0,0 +1,71 @@
+function [obj,al_info,alatt,angdeg] = remove_legacy_alignment(obj,varargin)
+%REMOVE_LEGACY_ALIGNMENT:  modify crystal lattice and orientation matrix
+% to remove legacy alignment applied to the crystal earlier.
+% Inputs:
+% obj    -- legacy realigned dnd object. Algorithm throws if the object has
+%           not been realigned using legacy algorithm.
+% Optional
+% alatt   -- lattice parameters with new values for lattice. (presumably
+%            before alignment)
+% angdeg  -- lattice angles with values necessary to set.
+%            Of one is present, another one have to be present.
+%            If these values are missing, assumes that the lattice have not
+%            been changed.
+% Outputs:
+% wout    -- dealigned dnd object
+% al_info -- instance of crystal_alignment_info class, containing alignment
+%            parameters, used to do legacy alignment.
+% Two input parameters duplicate the input provided in varargin (for runing realignment within the sqw object):
+% alatt   -- lattice parameters with new values for lattice. (presumably
+%            before alignment)
+% angdeg  -- lattice angles with values necessary to set.
+%            Of one is present, another one have to be present.
+%            If these values are missing, assumes that the lattice have not
+%            been changed.
+%
+%
+% Throws invalid_argument if identified that crystal has not been realigned
+
+if nargin > 1
+    if nargin ~= 3
+        error('HORACE:DnDBase:invalid_argument', ...
+            'If lattice parameters are provided as input (%s) lattice angles have to be provieded too. Got %d input parameters', ...
+            disp2str(varargin{1}),nargin);
+    end
+    % use tmp proj class to check the lattice parameters validity
+    tmp_proj = ortho_proj('alatt',varargin{1},'angdeg',varargin{2});
+    alatt = tmp_proj.alatt;
+    angdeg = tmp_proj.angdeg;
+    if any(abs(alatt-obj.alatt)>4*eps('single')) || ...
+            any(abs(angdeg-obj.angdeg)>4*eps('single'))
```

Comment:
```suggestion
    if ~equal_to_tol(alatt, obj.alatt, 'tol', 4*eps('single')) || ...
        ~equal_to_tol(angdeg, obj.angdeg, 'tol', 4*eps('single'))
```

Reply:
`equalt_to_tol` is complex function which analyzes various input parameters types and forms proper message which allows to identify the difference. Its primary usage is testing. Despite its not a performance critical part of code, I would rather not use it for simple check within workflow.